### PR TITLE
Migrated to th2 gradle plugin and Kotlin DSL

### PR DESCRIPTION
**Pull request by Nikita-Smirnov-Exactpro**

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,17 @@
+plugins {
+    kotlin("jvm") version "1.8.22" apply false
+    id("com.google.protobuf") version "0.9.3" apply false
+    id("com.exactpro.th2.gradle.grpc") version "0.0.3" apply false
+    id("com.exactpro.th2.gradle.base") version "0.0.3"
```
Comment (OptimumCode) [inline build.gradle.kts:5]: suggestion: I think it would be better if we put plugins we apply first and then all other plugins

File: core/build.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,91 @@
+plugins {
+    id("org.jetbrains.kotlin.jvm")
+    id("com.google.protobuf")
```
Comment (OptimumCode) [inline core/build.gradle.kts:3]: issue: Why do you need protobuf plugin in core module? I think we should remove it as this mode does not have ane gRPC

File: core/build.gradle.kts [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+
+configurations.all {
+    resolutionStrategy.cacheChangingModulesFor(0, "seconds")
+    resolutionStrategy.cacheDynamicVersionsFor(0, "seconds")
+}
+
+dependencies {
+    implementation(project(":grpc-read-db"))
+
+    api(platform("com.exactpro.th2:bom:4.5.0"))
```
Comment (OptimumCode) [inline core/build.gradle.kts:27]: issue: BOM will be added automatically by base th2 gradle plugin

**Review by OptimumCode — CHANGES_REQUESTED** [review]
I am not sure but probably we should check setup.py if it requires since the location of proto files has been changed

**Comment by Nikita-Smirnov-Exactpro** [comment]
> I am not sure but probably we should check setup.py if it requires since the location of proto files has been changed

setup.py looks at the 'build/generated/source/proto/main/services/python' folder where gradle puts generated code. I didn't find references to the source proto file

**Comment by Nikita-Smirnov-Exactpro** [comment]
Hm ... I don't understand why full build isn't triggered

**Comment by OptimumCode** [comment]
> Hm ... I don't understand why full build isn't triggered

Because there is an error in workflow

```
Invalid workflow file: .github/workflows/dev-build.yml#L35
The workflow is not valid. .github/workflows/dev-build.yml (Line: 35, Col: 20): Invalid secret, nvd-api-key is not defined in the referenced workflow.
```

File: gradle.properties [hunk 4]
```diff
@@ -1,3 +1,3 @@
 vcs_url=https://github.com/th2-net/th2-read-db
 release_version=0.0.0
-version_suffix=
\ No newline at end of file
+version_suffix=
```
Comment (OptimumCode) [inline gradle.properties:3]: issue (non-blocking): Please, revert this change. No need to change the file in PR if we don't need those changes

**Comment by OptimumCode** [comment]
Seems like the python build uses `proto` src after all
`error: cannot copy tree 'proto/th2_grpc_read_db': not a directory`

**Comment by OptimumCode** [comment]
It happens [here](https://github.com/th2-net/th2-read-db/blob/migrate-to-plugin/grpc/setup.py#L40)

File: core/build.gradle.kts [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+    id("java-library")
+    id("maven-publish")
+}
+
+val coroutinesVersion by extra("1.7.3")
+
+java {
+    withJavadocJar()
+    withSourcesJar()
+}
```
Comment (OptimumCode) [inline core/build.gradle.kts:12]: note: this will be applied automatically by th2 publish plugin

**Review by OptimumCode — APPROVED** [review]

**Review by OptimumCode — APPROVED** [review]
